### ISSUE TITLE
[$50] Community options Turned OFF again when user; turn ON the next item while the popup is there

### ISSUE THREAD

**Issue by sdgun** [topcoder-platform/community-app#1378]
## Steps to Reproduce

1. Open the Application and Login as a valid user
2. Go to Settings > Profile > Community
3. Turn OFF both the items and wait sometime
4. Now Turn On one item
5. When the popup is displayed; turn On the other one and observe
6. Repeat step 4-5 few times

## Actual Result(s)

- Community options Turned OFF again when user; turn ON the next item while the popup is there

## Expected Result(s)

- Must turn ON when user tap on it

## Frequency

- 5/5

## Environment Details

- **Device(s):** iPad Air 2
- **Operating System:** iOS 11.4.1
- **Size:** 9.7 inches (~71.6% screen-to-body ratio)
- **Resolution:** 1536 x 2048 pixels (~264 ppi pixel density)
- **Browser(s):** Safari 11

## Image(s)/Video(s)/Console Log/Crash Log

> If attachment(s) is/are not here, Please check the comments section 

## Ref:

@ThomasKranitsas

**Comment by sdgun** [comment]
In Desktop also

https://www.screencast.com/t/uVd5x2Xq

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30072958 has been created for this ticket.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30072958 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by ThomasKranitsas** [comment]
@topcoder-platform/topcodercompetitors this is open for pickup

**Comment by gets0ul** [comment]
@ThomasKranitsas Is this still valid? Are you able to reproduce it?

**Comment by ThomasKranitsas** [comment]
Yes, I was able to reproduce it today. I’ll attach a screen recording as soon as I can.

**Comment by gets0ul** [comment]
Thanks

**Comment by gets0ul** [comment]
I'm unable to reproduce it so I'm releasing this back to open.

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30072958 has been updated - it has been assigned to daga_sumit.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by sumitdaga** [comment]
@ThomasKranitsas 
i can't see any popup as of now when user toggles on any community!

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30072958 has been updated - it has been assigned to gurmeetb.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by gurmeetb** [comment]
@ThomasKranitsas let me give a shot and update in few hrs

**Comment by gurmeetb** [comment]
@ThomasKranitsas I see the problem. It is all because we are delaying the click handler by 1000 ms, so what happens is when we first switch off first one, and then by the time we do second one, the control has reached back componentWillReceiveProps(), which refreshes it. 

I did a test with removing the delay and it worked fine. Tried several times - didn't see issue then. 

But before I push the fix with disabling delay in update, I wanted to understand what was the idea behind setting a delay of 1000 ms or was it mistakenly done by the developer ?
